### PR TITLE
IA-1969 Add unit test for getAllInstanceNames

### PR DESCRIPTION
**Pull request by Qi77Qi**
**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

**Review by jdcanas — APPROVED** [review]

File: google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleDataprocInterpreterSpec.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.dsde.workbench.google2
+
+import com.google.cloud.dataproc.v1.{Cluster, ClusterConfig, InstanceGroupConfig}
+import org.broadinstitute.dsde.workbench.util2.{PropertyBasedTesting, WorkbenchTestSuite}
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import scala.collection.JavaConverters._
+
+class GoogleDataprocInterpreterSpec
```
Comment (kyuksel) [inline google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleDataprocInterpreterSpec.scala:9]: `final`?

**Review by rtitle — APPROVED** [review]

**Review by kyuksel — COMMENTED** [review]
Perhaps add a test for the _no instances_ case, too?

File: google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleDataprocInterpreterSpec.scala (hunk 1 above)
Comment (Qi77Qi) [reply google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleDataprocInterpreterSpec.scala:9]: I don't think we normally add `final` to spec files...

**Comment by Qi77Qi** [comment]
@kyuksel I think there will always be a master instance...but yea, in theory this will return empty map if no instances are defined in the fake `Cluster`

**Pull request merged**